### PR TITLE
Refactor module to use chain-registry for defaults based on chain-name

### LINKED ISSUE #4: Feat: Generalize osmosis-node module to cosmos-sdk-node using chain-registry
Seems like the `osmosis-node` module would naturally generalize to also work for other Cosmos-SDK based chains, using https://github.com/cosmos/chain-registry.

I'm imagining a workflow where you can specify the name of a chain (such as "osmosis"), and the module will use information from https://github.com/cosmos/chain-registry/blob/master/osmosis/chain.json to configure the setup process. (software repo, executable and folder names, chain-id, software version, seeds, etc)

This would mostly be useful for other networks wanting to reuse our code to run testnets, or infrastructure providers wanting to make mainnet nodes, but we could also use it for more realistic IBC testing.

### PR DESCRIPTION
**Pull request by UnityChaos**
Closes #4 

Implemented a basic setup of this idea, which uses the `chain-name` variable and https://github.com/cosmos/chain-registry to set default variables and parameters which can be overridden by manually setting terraform variables.

I also simplified the setup scripts in the module, to just install the chosen node software and initialize a genesis with a single validator. Would like to eventually extend that for other setups  (see #3), but I think that should be sorted out separately.

I'm still new to IAC, so feel free to advise if some of what I'm doing is hacky or non-standard

### CODE REVIEW

**Comment by ValarDragon** [comment]
As a headsup, we will be changing app.toml for Osmosis, and config.toml defaults, so we may still need to make osmosis start scripts. (But perhaps it can invoke the general one, and then adapt things after?)

File: .gitignore [hunk 1]
```diff
@@ -1 +1 @@
-/.terraform/
+*terraform*
```
Comment (nikever) [inline .gitignore:1]: I think this is too wide. The main goal was to block any `.terraform` folder. I would change it to `**/.terraform/`

File: modules/digital-ocean/cosmos-sdk-node/node.tf [hunk 2]
```diff
@@ -0,0 +1,45 @@
+locals {
+  username = "terraform"
```
Comment (nikever) [inline modules/digital-ocean/cosmos-sdk-node/node.tf:2]: Instead of `terraform`, maybe use something like `cosmos`?

File: README.md [hunk 3]
```diff
@@ -46,6 +46,6 @@ Currently this repository contains the following modules:
 
 |      Name      | Provider |      Description      |                     Usage                     |
 | -------------- | -------- | --------------------- | -------------------------------------------- |
-| `Osmosis Node` | DO       | Setup osmosis node(s) | [link](./modules/digital-ocean/osmosis-node) |
+| `Cosmos-SDK Node` | DO       | Setup Cosmos-SDK node(s) | [link](./modules/digital-ocean/cosmos-sdk-node) |
```
Comment (nikever) [inline README.md:49]: Could you format the table? 
You could use: http://markdowntable.com/ or there are various extensions for common IDEs

File: examples/do-cosmos-sdk-node/README.md [hunk 4]
```diff
@@ -62,7 +62,7 @@ make key
 Connect to one node using the key and the IP address:
 
 ```bash
-ssh -i key osmosis@159.203.80.187
+ssh -i key terraform@159.203.80.187
```
Comment (nikever) [inline examples/do-cosmos-sdk-node/README.md:65]: As above, Instead of `terraform`, maybe use something like `cosmos`?

**Review by nikever — CHANGES_REQUESTED** [review]
LGTM. Great work. 
I like the idea of generalizing the module using the `chain-registry`

I have added some suggestions, more cosmetic changes than substantial ones.

One thing, I have used [terraform-docs](https://github.com/terraform-docs/terraform-docs) to automatically generate the documentation for the terraform module. Could you use it to update the module documentation? Sorry it wasn't really documented

**Pull request closed without merging**